app/models/event_schedule.rb

```diff
@@ -12,6 +12,7 @@ class EventSchedule < ActiveRecord::Base
   scope :confirmed, -> { joins(:event).where('state = ?', 'confirmed') }
   scope :canceled, -> { joins(:event).where('state = ?', 'canceled') }
   scope :withdrawn, -> { joins(:event).where('state = ?', 'withdrawn') }
+  scope :current, -> { where('start_time <= ?', Time.current).select{ |es| es.end_time >= Time.current }.compact }
```

Comment:
You don't need `compact` with select.

`.map{ |es| es if es.end_time >= Time.current }.compact`
is equal to
`.select{  |es| es.end_time >= Time.current }`
So that is what select is for, to return an array containing all elements for which the given block returns a true value.